test/persistence-hooks.suite.js

```diff
@@ -274,26 +274,29 @@ module.exports = function(dataSource, should) {
           });
       });
 
-      // TODO(bajtos) Enable this test for all connectors that
-      // provide optimized implementation of findOrCreate.
-      // The unoptimized implementation does not trigger the hook
-      // when an existing model was found.
-      it.skip('triggers `before save` hook when found', function(done) {
-        TestModel.observe('before save', pushContextAndNext());
-
-        TestModel.findOrCreate(
-          { where: { name: existingInstance.name } },
-          { name: existingInstance.name },
-          function(err, record, created) {
-            if (err) return done(err);
-            observedContexts.should.eql(aTestModelCtx({ instance: {
-              id: record.id,
-              name: existingInstance.name,
-              extra: undefined
-            }}));
-            done();
-          });
-      });
+      if (dataSource.connector.findOrCreate) {
+        it('triggers `before save` hook when found', function(done) {
+          TestModel.observe('before save', pushContextAndNext());
+
+          TestModel.findOrCreate(
+            { where: { name: existingInstance.name } },
+            { name: existingInstance.name },
+            function(err, record, created) {
+              if (err) return done(err);
+              observedContexts.should.not.eql(aTestModelCtx({ instance: {
+                id: record.id,
+                name: existingInstance.name,
+                extra: undefined
+              }}));
+              record.toObject(true).should.eql({
+                id: existingInstance.id,
+                name: existingInstance.name,
+                extra: undefined
+              });
```

Comment:
The purpose of this test is to test the values passed to the hook (`observedContexts`), not the value of the argument passed to `findOrCreate` callback (`record`).

The check of `record.toObject()` belongs to a different test suite, most likely `test/manipulation.test.js`.

It may be possible to modify the instance properties checked by `observedContexts.should.eql(aTestModelCtx({ instance: {` to make this test pass. I am ok with that as long is the change makes sense. However, it is more likely that the test is not passing because of a problem in the implementation of `findOrCreate`. Especially since an almost identical test `it triggers "before save" hook when not found` below is passing.